addon/types/model_export/model.py

```diff
@@ -172,14 +177,34 @@ def generate_qc(self):
             qc.write('$mostlyopaque')
             qc.write('\n')
 
+        if self.transform_source == 'MANUAL':
+            origin_x = self.origin_x
+            origin_y = self.origin_y
+            origin_z = self.origin_z
+            rotation = self.rotation
+            scale = self.scale
+        elif self.transform_source == 'OBJECT' and self.transform_object is not None:
+            matrix = self.transform_object.matrix_world.decompose()
+            origin_x = -matrix[0][0]
+            origin_y = -matrix[0][1]
+            origin_z = -matrix[0][2]
+            rotation = degrees(matrix[1].to_euler('XYZ')[2])
+            scale = matrix[2][2]
```

Comment:
Note: Read the whole comment before you start coding; it lists multiple options, don't want you to waste time implementing them all.

You my friend need to learn about tuple unpacking.
Instead of writing `matrix = something.decompose()`, do `loc, rot, sca = something.decompose()`.
The variable name you used here (`matrix`) isn't accurate, because it's not a matrix, it's a tuple with a vector, a quaternion, and another vector.

To get the `origin_x/y/z` you can do `origin_x = loc.x` (I'd put y and z on their own lines also).
To get the `rotation` you can do `rotation = degrees(rot.to_euler().z)`.
To get the `scale` you can do `scale = sca.x` (or y or z, I guess I'd go with z).

You may want to round the numbers to say 6 digits after the dot though cause float inaccuracy.
To do that it's for example `origin_x = round(loc.x, 6)` (the second number is digits after the dot).

Alternatively you could do it during conversion to string, like this:

        qc.write('\n')
        qc.write(f'$origin {origin_x:.6f} {origin_y:.6f} {origin_z:.6f} {rotation:.6f}')
        qc.write('\n')

        qc.write('\n')
        qc.write(f'$scale {scale:.6f}')
        qc.write('\n')

I'd probably go with that because then it affects manual input values too.

Reply:
ah, the `loc, rot, sca = something.decompose()` part makes sense now, I was trying it somewhere else but got an error about not being able to unpack a tuple or something like that, now I see what the syntax can be for, really neat!